std/uni.d

```diff
@@ -2541,10 +2541,14 @@ private:
 
     ref intersect()(dchar ch)
     {
-        foreach (i; byInterval)
+        scope arr = data;
+        foreach (i; Intervals!(typeof(data))(arr))
             if (i.a <= ch && ch < i.b)
-                return this = This.init.add(ch, ch+1);
-        this = This.init;
+            {
+                data.length(0);
```

Comment:
Nitpick: I think the intended usage is:
````
data.length = 0;
````

Reply:
I didn't go through all details of CowArray and Your comment was my first idea too, but as it is reference counted, I opted for the method
@property void length(size_t len)
which cares for refCount

Reply:
@quickfur 's suggestion still calls that method:

```
void length (int val)
{
    import std.stdio;
    writeln(val);
}

void main()
{
    length = 1;
}
```

(prints 1).

Reply:
I think just using return scope should be the ticket.

No idea what this muckering with scope here acomplishes but I think it just fools compiler

Reply:
@carblue I think you misunderstood. The syntax `data.length = 0;` is lowered to `data.length(0)`. It's not a big deal, but usually when you write `@property void length(size_t) { ... }`, the intention is to use it as a property setter method, i.e., you would invoke it with assignment syntax rather than function call syntax.

Reply:
Agreed about data.length = 0; and thanks, I forgot about that lowering.

@DmitryOlshansky The scope topic is tricky; I watched Walter Bright's Pointers Gone Wild: Memory Safety and D several times, introduced that into some code and know from tests, that the implementation still needs some refinement: I wouldn't call it "fool", but just "help" the compiler with it's current, hypersensitive dip1000 code and/or help me save time: Whether legitimately or not  (I didn't go to the bottom of that in static struct Intervals(Range) or wherever, which may be worthwhile), the compiler - in the first place - rejects byInterval in foreach with the error mentioned in issue 17961. The simple, yet maybe temporary solution I opted for, is, to give the compiler what it seemingly wants to see here, a scope variable (arr) -based iteration.
I hope, when dmd is dip1000-ready, the compiler will be that 'clever' to detect the helping hand case here, making arr dispensable.

If I understood Your ticket right, 'return scope this;' doesn't compile for syntax reason.